src/main/java/duke/utility/TaskList.java

```diff
@@ -0,0 +1,131 @@
+package duke.utility;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+
+/**
+ * <h2>TaskList</h2>
+ * Organises all tasks created by the user.
+ * Handles details from the {@link duke.utility.Parser} class and adds, modifies, or deletes tasks accordingly.
+ */
+
+public class TaskList {
+
+    protected final HashSet<String> existingTasks;
+    private final List<Task> tasks;
```

Comment:
Maybe the variable name tasks could be made more specific to describe what is being stored in the List, because having it might be unclear what this variable is being used for especially since it is placed under another variable with a similar name.